internal/clients/e2e.go

```diff
@@ -269,7 +269,7 @@ func RunEndToEnd(ctx context.Context, cfg *config.E2ERunnerConfig) error {
 			}
 			publishResp, err := keyServerClient.Publish(ctx, publishReq)
 			defer logger.Debugw("publish", "request", publishReq, "response", publishResp)
-			if publishResp.ErrorMessage != "" {
+			if err != nil || publishResp.ErrorMessage != "" {
```

Comment:
I think we need to take an approach like L244 and return a different error structure